src/components/LoginForm/LoginForm.jsx

```diff
@@ -0,0 +1,130 @@
+                                        className="mb-2" 
+                                        type={inputs.type} 
+                                        name={inputs.name} 
+                                        id={inputs.name} 
+                                        placeholder ={inputs.placeholder} />       
+                        </FormGroup>  
+                        )
+                    }
+                </div>
+                <FormGroup>
+                    <Button size="lg" pill theme="dark" onClick= {(event) => handleButtonClick(event)}  name="submit">{"Login"}</Button> 
+                </FormGroup>
+            </Form>
+        </div>
+    );
+};
+
+export default (props) => {
+    const defaultErrors = {
+        email:      {    
+                        hasError:false,
+                        message: "" 
+                    },
+        password:   {   hasError:false,
+                        message: "" 
+                    }   
+    };
+
+    const [user_credentials, setUserCredential] = useState({}); 
+    const [error_message, setErrorMessage] = useState("");
+    const [hasErrors, setHasErrors] = useState(defaultErrors);
+    const [alertMessage, setAlertMessageFields] = useState({
+        visible: false,
+        alertType: "danger"
+    });
+
+    /**
+     * Triggers the error alert banner.
+     * !!!! Temporary put a success alert for successfull login. 
+     * @param {String} alertType the alert type success or error.
+     */
+    const triggerAlert = (alertType) => {
+        setAlertMessageFields({visible: !alertMessage.visible,
+                               alertType: [alertType]});
+    };
+    /**
+     * @function handleResponse
+     * @description filters the error type and setting the erro message and form tip error.
+     * @param {Object} response the Http response from backend.
+     */
+   const handleResponse = (response) => {
+        if(response.status === "BAD_REQUEST"){
+            let errors = response.errors;
+            errors.map(error => (
+                setHasErrors({...hasErrors, [error.field]: {...hasErrors[error.field], hasError: true , message: error.message}}) 
+            ))
+        };
+  
+        return setErrorMessage(response.message);
+    };
+    /**
+     * @function handleButtonClick 
+     * @description Handles the submission, calls the backend API, through login function.
+     * handle the response with handleResponse function.
+     * @param {Event} event 
+     */
+    const handleButtonClick = (event) => {
+        event.preventDefault();
+        login(user_credentials).then(response => {
+            if(!response.token) { 
+                triggerAlert("danger");
+                handleResponse(response);
+            }
+            else {
+                localStorage.setItem("token", response.token);
+                triggerAlert("success");//temporary TODO: will be redirect to home. props.history.push('/home')
```

Comment:
`this.props.history.push("/dashboard");`

https://dev.to/projectescape/programmatic-navigation-in-react-3p1l